library/Businessprocess/HostNode.php

```diff
@@ -56,8 +58,18 @@ public function getUrl()
             'host' => $this->getHostname(),
         );
 
-        if ($this->getBpConfig()->hasBackendName()) {
-            $params['backend'] = $this->getBpConfig()->getBackendName();
+        if ($this->getBpConfig()->hasBackendName() ||
+            (Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend())
+        ) {
+            $backendName = $this->getBpConfig()->getBackendName();
+
+            if (Module::exists('icingadb') &&
+                ($backendName === '_icingadb' || IcingadbSupport::useIcingaDbAsBackend())
+            ) {
+                $params['backend'] = '_icingadb';
+            } else {
+                $params['backend'] = $this->getBpConfig()->getBackendName();
+            }
```

Comment:
> * Explicit backend?
>   * Add paramter backend

Still doesn't happen. if `$this->getBpConfig()->hasBackendName()` is true **and** `Module::exists('icingadb') && IcingadbSupport::useIcingaDbAsBackend()` (on lines 66-67), the real backend name is still ignored.

> * Default backend?
>   * Let the target decide, just like it did the entire time prior this by assuming the default monitoring backend if there is no parameter named backend

Also doesn't happen. Why do you add `_icingadb` as backend? Let the `HostController::showAction` and `ServiceController::showAction` decide what to do if there's no `backend` parameter instead. (i.e. No `backend` parameter? Is Icingadb enabled and the preferred backend? Use it! Monitoring otherwise.)

Note that you don't even need to change anything here for this solution.